circuit_maintenance_parser/processor.py

```diff
@@ -117,16 +117,27 @@ class CombinedProcessor(GenericProcessor):
 
     def process_hook(self, maintenances_extracted_data, maintenances_data):
         """All the parsers contribute with a subset of data that is extended."""
-        # We only expect one data object from these parsers
         if len(maintenances_extracted_data) == 1:
             self.combined_maintenance_data.update(maintenances_extracted_data[0])
         else:
-            raise ProcessorError(f"Unexpected data retrieved from parser: {maintenances_extracted_data}")
+            maintenances_data.extend(maintenances_extracted_data)
```

Comment:
I believe it would be useful to document the new logic in the docstring.
To make clear that when the parser return one object we just accumulate it in the `combined_maintenance_data` but when it's multiple, we understand that there are multiple `maintenances` so we save them in the `maintenances_data` to be all extended at the end.
Maybe we could add a safety check if we try this `maintenances_data.extend` twice... this should only happen one time in this processor